File: leetcode/leetcode_py/find-the-highest-altitude.py
'''
https://leetcode.com/problems/find-the-highest-altitude

1. Iterate and accumulate
time: o(n)
space: o(1)
'''
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        res = 0
        cur = 0
        for val in gain:
            cur += val
            res = max(cur, res)
        return res

        

File: leetcode/leetcode_cpp/find-the-highest-altitude.cpp
/*
https://leetcode.com/problems/find-the-highest-altitude

1. Iterate and accumulate
time: o(n)
space: o(1)
*/
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int res = 0;
        int cur = 0;
        for (int i=0; i<gain.size(); ++i) {
            cur += gain[i];
            res = max(res, cur);
        }
        return res;
    }
};